hibernate-testing/src/main/java/org/hibernate/testing/hamcrest/InitializationCheckMatcher.java

```diff
@@ -42,12 +42,16 @@ public InitializationCheckMatcher(boolean assertInitialized) {
 
 	@Override
 	public boolean matches(Object item) {
-		final boolean initialized = Hibernate.isInitialized( item );
-		return assertInitialized ? initialized : !initialized;
+		return assertInitialized == Hibernate.isInitialized( item );
 	}
 
 	@Override
 	public void describeTo(Description description) {
-		description.appendValue( "Hibernate#isInitialized == " + !assertInitialized );
+		description.appendValue( "Hibernate#isInitialized() returns " + assertInitialized );
```

Comment:
The above method will be used for 'Expected', so the original code logic is incorrect. The 'actual' counterpart is customized below.